wc3-hw1-TranNgocLam/cypress/e2e/registrationForm.cy.js

```diff
@@ -0,0 +1,308 @@
+        .checkGender(users[0].gender)
+        .inputUserNumber(users[0].userNumber)
+        .chooseDOB(users[0].dob_day, users[0].dob_month, users[0].dob_year)
+        .inputSubjects(users[0].subjects)
+        .checkHobbies(users[0].hobbies)
+        .uploadPicture(users[0].picture)
+        .inputCurrentAddress(users[0].currentAddress)
+        .inputState(users[0].state)
+        .inputCity(users[0].city)
+        .clickSubmit()
+
+      confirmForm
+        .isCorrectName(users[0].firstName, users[0].lastName)
+        .isCorrectGender(users[0].gender)
+        .isCorrectEmail(users[0].email)
+        .isCorrectMobile(users[0].userNumber)
+        .isCorrectDOB(users[0].dob_day, users[0].dob_month, users[0].dob_year)
+        .isCorrectSubjects(users[0].subjects)
+        .isCorrectHobbies(users[0].hobbies)
+        .isCorrectPicture(users[0].picture)
+        .isCorrectAddress(users[0].currentAddress)
+        .isCorrectStateAndCity(users[0].state, users[0].city)
+        .clickClose()
+    });
+
+  });
+
+  // TCRegister#2: Verify that user can submit successfully with valid data in required data fields
+  it('TCRegister#2', () => {
+    cy.get('@registration').then((users) => {
+
+      registerForm
+        .inputLastName(users[1].lastName)
+        .inputFirstName(users[1].firstName)
+        .checkGender(users[1].gender)
+        .inputUserNumber(users[1].userNumber)
+        .clickSubmit()
+
+      confirmForm
+        .isCorrectName(users[1].firstName, users[1].lastName)
+        .isCorrectGender(users[1].gender)
+        .isCorrectMobile(users[1].userNumber)
+        .clickClose()
+    });
+  });
+
+  /*   TCRegister#3: Verify that user submit successfully when first name, last name, current adrress have abundant space  at the begin and end of input data:
+     - First name, Last name, Current Adress have more than 255 characters.
+     - Picture file's size exceeds limit file size 2MB. */
+  it('TCRegister#3', () => {
+    cy.get('@registration').then((users) => {
+
+      registerForm
+        .inputLastName(users[0].lastName)
+        .inputFirstName(users[0].firstName)
+        .checkGender(users[0].gender)
+        .inputUserNumber(users[0].userNumber)
+        .clickSubmit()
+
+      confirmForm
+        .isCorrectName(users[0].firstName, users[0].lastName)
+        .isCorrectGender(users[0].gender)
+        .isCorrectMobile(users[0].userNumber)
+        .clickClose()
+    });
+  });
+
+  // TCRegister#4: Verify that user cannot submit with empty data fields.
+  it('TCRegister#4', () => {
+    registerForm
+      .clickSubmit()
+      .checkFirstNameValid(false)
+      .checkLastNameValid(false)
+      .isGenderChecked(false)
+      .checkMobileValid(false)
+  });
+
+    /*  TCRegister#5: Verify that user cannot submit with invalid data:
+      - First Name includes numbers and special characters.
+      - Last Name includes numbers and special characters */
+    it.only('TCRegister#5', () => {
```

Comment:
Remove `only` when commit, this case has same steps with second case, you should make a list of data and loop through it, so you just have to write these boring lines once